srcs/app/http/response/delete/HttpDeleteResponse.cpp

```diff
@@ -3,18 +3,64 @@
 HttpDeleteResponse::HttpDeleteResponse(
 	RequestConfig *requestConfig,
 	HttpRequest *request) : HttpBaseResponse(requestConfig, request) {
-	// TODO(any) Implement DELETE
 	if (error_code_) {
-		raw_response_ = HttpErrorResponse(
-										error_code_,
-										request_config_,
-										request_).Content();
+		SetErrorRawResponse_(error_code_);
 	} else {
-		HttpResponse::HeadersMap headers;
-		std::string	body = "Responding to a DELETE request\n";
+		const std::string full_path =
+				request_config_->GetRoot() + request_->GetDecodedPath();
+		try {
+			File file(full_path);
 
-		headers.insert(std::make_pair("Content-Type", "text/plain"));
-		headers.insert(std::make_pair("Location", request_config_->GetReturnUrl()));
-		SetRawResponse_(request_config_->GetReturnStatus(), headers, body);
+			if (file.IsRegularFile() || file.HasEndSlash()) {
+				Delete_(file);
+			} else {
+				SetErrorRawResponse_(409);
+			}
+		} catch(File::Error & e) {
+			SetErrorRawResponse_(e.what());
+		}
 	}
 }
+
+void	HttpDeleteResponse::Delete_(File const &file) {
+	if (file.IsDirectory()) {
+		BuildResponse_(
+			nftw(file.GetPath().c_str(),
```

Comment:
So the idea behind this ntfw is to remove all contents in a directory?

Reply:
Yes, I also though about it. Since `nftw` is a function which main purpose is to traverse the directory tree and applying a function (in this case `RemoveSingleFile_`) for me was more legible to distinguish between the two cases: one having to walk through al the file hierarchy and then a case for deleting simple file. But yes, it should also work :)